app/helpers/course/assessment/submissions_helper.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+module Course::Assessment::SubmissionsHelper
+  # The maximum step that current user can attempt.
+  def max_step
+    @max_step ||= @assessment.
+                  questions.index(@assessment.questions.next_unanswered(@submission)) + 1
+  end
+
+  # The step that current user is on.
+  def current_step
+    @current_step ||= begin
+      current_question = @questions_to_attempt.first
```

Comment:
Then the user is not able to step to next question,  `@questions_to_attempt` will be his previous question.